src/popover/internal.tsx

```diff
@@ -99,6 +102,19 @@ function InternalPopover(
     };
   }, []);
 
+  useEffect(() => {
+    if (
+      visible === false &&
+      wasVisible === true &&
+      document.activeElement === document.body &&
+      !dismissedViaDocumentClick
+    ) {
+      triggerRef.current?.focus();
```

Comment:
It is possible to do it the moment dismiss happens? E.g. when the dismiss button is pressed or Esc keypress is intercepted?

Having this as a side-effect often leads to tricky issues.

Reply:
Or maybe it is possible to change onDismiss:

```
const onDismiss = useCallback(() => {
    setVisible(false);
    if (shouldFocusTrigger()) {
      triggerRef.current?.focus();
    }
}, []);
```

The shouldFocusTrigger part can check the last interaction -- if it was mouse-down or TAB keypress -- ignore, otherwise -- focus.

Reply:
Seems a good simplification, taking a look